src/certs.rs

```diff
@@ -187,32 +105,146 @@ pub(crate) async fn create_tls_config_and_watch_certificate_changes(
                 key_changed = true;
             }
 
-            for client_cert_watch in client_cert_watches.iter() {
-                if event.wd == *client_cert_watch {
+            for client_ca_watch in client_ca_watches.iter() {
+                if event.wd == *client_ca_watch {
                     info!("TLS client certificate file has been modified");
-                    client_cert_changed = true;
+                    client_ca_changed = true;
                 }
             }
 
-            // if both the certificate and the key have been changed or there is no change in the
-            // server cert and key, but the client cert changed, reload the certificate
-            if (key_changed && cert_changed)
-                || (client_cert_changed && (key_changed == cert_changed))
-            {
-                info!("Reloading TLS certificates");
+            // Reload the client CA certificates if they have changed, keeping the current server certificates unchanged
+            if client_ca_changed {
```

Comment:
If an inotify event indicates a change in the client CA, only the client verifier should be reloaded. If the TLS server certificate or key were modified simultaneously, but the inotify event for those changes had not yet arrived, we could end up with a broken TLS configuration till the next inotify event. Reloading only the client verifier in this scenario avoids partial updates of the certificate and key.